doc/api/http.md

```diff
@@ -178,6 +186,20 @@ changes:
     **Default:** `'lifo'`.
   * `timeout` {number} Socket timeout in milliseconds.
     This will set the timeout when the socket is created.
+  * `proxyEnv` {Object|undefined} Environment variables for proxy configuration.
+    See [Built-in Proxy Support][] for details. **Default:** `undefined`
+    * `HTTP_PROXY` {string|undefined} URL for the proxy server that HTTP requests should use.
+      If undefined, no proxy is used for HTTP requests.
+    * `HTTPS_PROXY` {string|undefined} URL for the proxy server that HTTPS requests should use.
+      If undefined, no proxy is used for HTTPS requests.
+    * `NO_PROXY` {string|undefined} Patterns specifying the endpoints
+      that should not be routed through a proxy.
+    * `http_proxy` {string|undefined} Same as `HTTP_PROXY`. If both are set, `http_proxy` takes precedence.
+    * `https_proxy` {string|undefined} Same as `HTTPS_PROXY`. If both are set, `https_proxy` takes precedence.
+    * `no_proxy` {string|undefined} Same as `NO_PROXY`. If both are set, `no_proxy` takes precedence.
```

Comment:
It makes the code to check for these a bit more complicated but could we instead just say that the keys are case-insensitive and only document one set of these?

Reply:
Then let's only support the upper-case variants here. The `process.env` does not treat environment variables this way, e.g. `process.env.foo` and `process.env.FOO` are two entirely different env vars

Reply:
According to https://about.gitlab.com/blog/we-need-to-talk-no-proxy/#http_proxy-and-https_proxy if we only want to support one, lower-cased one is probably the best, supporting the upper-case only variant is more likely to lead to woes to those who administer the systems (not necessarily Node.js developers).

I think this is special in that, we are supporting an ecosystem convention instead of just inventing our own, so we'd need to give in to what's more common in the ecosystem out there to avoid introducing Node.js-only quirks, or at least a common denominator.

Reply:
I still generally think supporting both the uppercase and lowercase variants here is a DX mistake but don't feel strongly enough to block.